Framework/HistogramData/inc/MantidHistogramData/VectorOf.h

```diff
@@ -116,7 +116,14 @@ template <class T, class CowType> class VectorOf {
       m_data = Kernel::make_cow<CowType>(std::move(data));
     return *this;
   }
-
+  /// Assigns the contents of the range [first, last]
+  template <class InputIt> void assign(InputIt first, InputIt last) & {
+    m_data.access().assign(first, last);
+  }
+  /// Assigns data of size count with all elements initialized to value.
+  void assign(size_t count, const double &value) & {
+    m_data.access().assign(count, value);
+  }
```

Comment:
Good.
Could you also add tests for these and those in `FixedLengthVector`? `HistogramData` is a low level library that is used everywhere, so I am trying to be really thorough and test everything.